src/shared/firewall-util-nft.c

```diff
@@ -812,6 +839,66 @@ static int fw_nftables_recreate_table(sd_netlink *nfnl, int af, sd_netlink_messa
         return 0;
 }
 
+static int nft_message_add_setelem_ip6range(sd_netlink_message *m,
+                                            const union in_addr_union *source,
+                                            unsigned int prefixlen) {
+        struct in6_addr start, end;
+        uint8_t carry = 0;
+        int r, i, j;
+
+        assert(prefixlen <= 128);
+
+        for (i = 0, j = 15; i < 16; i++) {
+                uint8_t nm;
+
+                nm = 0xFF;
+                if (prefixlen < 8)
+                        nm = 0xFF << (8 - prefixlen);
```

Comment:
`prefixlen` maybe `8`...
```
if (prefixlen  == 0)
     nm = 0;
else if (prefixlen < 8)
    nm <<= (8 - prefixlen)
```

Reply:
Not following, sorry. If prefixlen is 8, nm should remain at 0xff. If prefixlen is 0, then nm should be 0.
AFAICS this is the case (0xff << 8 is reduced to 0 on assignment).